docs/src/scratchspaces.md

```diff
@@ -0,0 +1,150 @@
+# [**9.** Scratch Spaces](@id Scratch-Spaces)
+
+!!! compat "Julia 1.6"
+    Pkg's Scratch Spaces functionality requires at least Julia 1.6.
+
+`Pkg` can manage and automatically garbage collect scratch spaces of temporary or readily-recreatable data.
```

Comment:
> temporary or readily-recreatable

This phrasing makes me a bit worried for using it for something like conda environments. But, IIUC, as long as the relevant `Project.toml` exists, the scratch space won't be GC'ed?

Reply:
No, in the current implementation, if a package doesn't access it often enough, it will be reaped.

Perhaps this isn't how we want it to be done, and we instead want it to exist for as long as the owning Project.toml exists.  I'm open to either interpretation, but the use cases I have in mind are always such that it's okay that the underlying scratch space gets emptied every now and then. (They were originally called "caches", but we decided the name was too generic)

If you have something that should persist in a stronger way, I suggest just using a folder on-disk somewhere.

Reply:
Thanks for the explanation. I guess it makes sense to eagerly GC things in auto-generated non-human-readable paths.

Reply:
The default collection policy here is to wait 21 days after the last `@get_scratch!()` that targeted it, then move it to an orphanage and then clean it with everything else a week later, so on average it should take 3-4 weeks for something to get GC'ed.  I figured this was probably generous enough.

Reply:
One thing that is not 100% clear to me from the Pkg.jl documentation: are garbage collections only triggered by the user explicitly asking for it, or also in other circumstances?

Reply:
Right now, it's only when users manually run `gc()`.  In the future, we may run `gc()` automatically.